File: langs/python/utils/to_string.py
# coding: utf-8

def sSs(args):
  res = "["; lim = len(args) - 1
  for i in range(lim):
    res += "'" + args[i] + "'" + ", "
  res += "'" + args[lim] + "']"
  return res
def test_sSs(): res = sSs(["hoge","fuga","mamo"]); print(res)   #exp is ['1', '2', '3']

def sNs(args):
  return sSs([str(arg) for arg in args])
def test_sNs(): res = sNs([1,2,3]); print(res)   #exp is ['1', '2', '3']

def to_ss(ns): return [str(n) for n in ns]
def unwords(ns): ss = to_ss(ns); return " ".join(ss)
def test_unwords(): res = unwords([1,2,3]); print(res)

if __name__ == "__main__":
  test_sSs()
  test_sNs()
  test_unwords()

File: langs/ruby/utils/to_string.rb
#encoding: UTF-8

def sNs(ns)
  res = '['; lim = ns.size()-1;
  lim.times{|i| res += '"' + ns[i].to_s + '", '}
  res += '"' + ns[lim].to_s + '"]'
  return res
end
def test() res = sNs([3]); puts(res) end   #exp is ["1","2","3"]

if __FILE__ == $0 then
  test
end